app/controllers/registrations_controller.rb

```diff
@@ -4,11 +4,33 @@ class RegistrationsController < Devise::RegistrationsController
   layout :layout_from_site_context
 
   def create
-     super do |user|
-       if user.persisted?
-         BootstrapUser.(user, session[:user_join_track_id])
-       end
-     end
+    build_resource(sign_up_params)
```

Comment:
I would rather not override the whole `create` method if we can avoid it, as that means we have to update it whenever we change Devise. 

Because you've made the mail sending asynchronous below, I would have thought that this method will now never raise the `ApplicationMailer::UnableToConnectToSMTPServerError`, so the controller doesn't require any changes. Thoughts?

Reply:
Good point. It should raise the error outside the controller if that was the case. I'll look into why in our tests in raises it inline.